### PR TITLE
Daedalus changes for MWS

### PR DESCRIPTION
**Pull request by tonypizzicato**

### CODE REVIEW

File: src/addressindex.h [hunk 1]
```diff
@@ -297,13 +294,15 @@ struct CMempoolAddressDeltaKey
...
+    CMempoolAddressDeltaKey(int addressType, uint160 addressHash, uint256 hash, unsigned int i, int s, bool is_invite) {
         type = addressType;
         addressBytes = addressHash;
         txhash = hash;
         index = i;
         spending = s;
+        invite = is_invite;
     }
 
     CMempoolAddressDeltaKey(int addressType, uint160 addressHash) {
```
Comment (mempko) [inline src/addressindex.h]: you should set invite here too

File: src/miner.cpp [hunk 2]
```diff
@@ -893,7 +893,7 @@ static bool ProcessBlockFound(const CBlock* pblock, const CChainParams& chainpar
 
     // Process this block the same as if we had received it from another node
     std::shared_ptr<const CBlock> shared_pblock = std::make_shared<const CBlock>(*pblock);
-    if (!ProcessNewBlock(chainparams, shared_pblock, true, nullptr, true))
+    if (!ProcessNewBlock(chainparams, shared_pblock, true, nullptr, false))
```
Comment (mempko) [inline src/miner.cpp]: You are turning off validation here, is it because it has already been validated?

File: src/refmempool.cpp [hunk 3]
```diff
@@ -30,6 +30,11 @@ std::string GetAlias(const RefMemPoolEntry& entry)
     return entry.GetSharedEntryValue()->alias;
 }
 
+Address GetParentAddress(const RefMemPoolEntry& entry)
```
Comment (mempko) [inline src/refmempool.cpp]: the return type can be a const reference here right?

File: src/rpc/misc.cpp [hunk 4]
```diff
@@ -801,6 +806,7 @@ UniValue getaddressmempool(const JSONRPCRequest& request)
             "    \"timestamp\"  (number) The time the transaction entered the mempool (seconds)\n"
             "    \"prevtxid\"  (string) The previous txid (if spending)\n"
             "    \"prevout\"  (string) The previous transaction output index (if spending)\n"
+            "    \"isInvite\"  (boolean) If transaction is an invite\n"
```
Comment (mempko) [inline src/rpc/misc.cpp]: should this be 'isinvite' seems inconsistent with other values.

File: src/rpc/misc.cpp [hunk 5]
```diff
@@ -839,6 +845,7 @@ UniValue getaddressmempool(const JSONRPCRequest& request)
         delta.push_back(Pair("satoshis", it->second.amount));
         delta.push_back(Pair("script", HexStr(it->second.scriptPubKey)));
         delta.push_back(Pair("timestamp", it->second.time));
+        delta.push_back(Pair("isInvite", it->first.invite));
```
Comment (mempko) [inline src/rpc/misc.cpp]: see above

File: src/rpc/misc.cpp [hunk 6]
```diff
@@ -1313,12 +1440,13 @@ template <class IndexPair>
 void DecorateEasySendTransactionInformation(UniValue& ret, const IndexPair& pair, const size_t& confirmations)
 {
     const auto& key = pair.first;
-    ret.push_back(Pair("found", true));
+
     ret.push_back(Pair("txid", key.txhash.GetHex()));
     ret.push_back(Pair("index", static_cast<int>(key.index)));
-    ret.push_back(Pair("amount", ValueFromAmount(GetAmount(pair.second))));
+    ret.push_back(Pair("amount", !key.invite ? ValueFromAmount(GetAmount(pair.second)) : GetAmount(pair.second)));
```
Comment (mempko) [inline src/rpc/misc.cpp]: doesn't the value 'pair.second' already have if it's an invite?

File: src/rpc/misc.cpp [hunk 7]
```diff
@@ -1313,12 +1440,13 @@ template <class IndexPair>
...
     ret.push_back(Pair("spending", key.spending));
     ret.push_back(Pair("confirmations", static_cast<int>(confirmations)));
+    ret.push_back(Pair("invite", key.invite));
```
Comment (mempko) [inline src/rpc/misc.cpp]: the value should have if it's an invite, not sure you need invite on the key.

File: src/miner.cpp (hunk 2 above)
Comment (tonypizzicato) [reply src/miner.cpp]: i've turned off sampling, not validation

File: src/rpc/misc.cpp (hunk 4 above)
Comment (tonypizzicato) [reply src/rpc/misc.cpp]: thought about it, but here's an `outputIndex` couple lines above
https://github.com/meritlabs/merit/blob/59a402867c267a338f907dc67f4ff8132223f9ea/src/rpc/misc.cpp#L804

File: src/rpc/misc.cpp (hunk 6 above)
Comment (tonypizzicato) [reply src/rpc/misc.cpp]: moved it to key as DecorateEasySendTransactionInformation uses key for polymorphic access to `invite` field in the key for both mempool index and address index
https://github.com/meritlabs/merit/blob/59a402867c267a338f907dc67f4ff8132223f9ea/src/rpc/misc.cpp#L1439
https://github.com/meritlabs/merit/blob/59a402867c267a338f907dc67f4ff8132223f9ea/src/rpc/misc.cpp#L1512

File: src/addressindex.h (hunk 1 above)
Comment (tonypizzicato) [reply src/addressindex.h]: only place it is used is here
https://github.com/meritlabs/merit/blob/59a402867c267a338f907dc67f4ff8132223f9ea/src/txmempool.cpp#L536
and it doesn't matter if it's an invite or not

File: src/rpc/misc.cpp (hunk 7 above)
Comment (tonypizzicato) [reply src/rpc/misc.cpp]: same here

File: src/miner.cpp (hunk 2 above)
Comment (mempko) [reply src/miner.cpp]: your right, the param name in header was misleading

**Pull request merged**